src/client/Microsoft.Identity.Client/Cache/Items/MsalAccessTokenCacheItem.cs

```diff
@@ -45,11 +48,26 @@ internal MsalAccessTokenCacheItem(
             RawClientInfo = response.ClientInfo;
             HomeAccountId = homeAccountId;
             OboCacheKey = oboCacheKey;
+            PersistedCacheParameters = AcquireCacheParametersFromResponse(additionalRequestRapameters, response.ExtensionData);
 
             InitCacheKey();
         }
 
-      
+        private IDictionary<string, string> AcquireCacheParametersFromResponse(
+                                                    IEnumerable<string> additionalRequestRapameters,
+#if SUPPORTS_SYSTEM_TEXT_JSON
+                                                    IDictionary<string, JsonElement> extraDataFromResponse)
+#else
+                                                    IDictionary<string, JToken> extraDataFromResponse)
+#endif
+        {
+            var cacheParameters = extraDataFromResponse
+                                    .Where(x => additionalRequestRapameters.Contains(x.Key))

```

Comment:
is there a case insensitive / culture agnostic version of Contains?

Reply:
I dont see it so not sure

Reply:
How about we don't use Linq here? This is in the hot path for MSAL. Just use foreach

Reply:
I found the ignore case for contains. So I think that is the better route to go, doing this with foreach requires a bit more logic.